packages/react-dom/src/shared/DOMProperty.js

```diff
@@ -206,6 +206,25 @@ export function getPropertyInfo(name: string): PropertyInfo | null {
   return properties.hasOwnProperty(name) ? properties[name] : null;
 }
 
+export function getCustomElementPropertyInfo(
+  name: string,
+  node: Element) {
+  if (name in (node: any)) {
+    const acceptsBooleans = (typeof (node: any)[name]) === 'boolean';
```

Comment:
Is this what preact does? What's the effective thing here? That if we pass null, we pass a boolean instead of empty string?

Seems like it's wrong either way, because it seems like we should pass null through to the property.

Reply:
Thats a good point.
I guess the `acceptsBooleans` and `type` flags in PropertyInfo don't really make sense for custom elements, do they...? Should we rethink using PropertyInfo for this feature and instead change all of the logic which uses PropertyInfo?

Reply:
> Is this what preact does? What's the effective thing here? That if we pass null, we pass a boolean instead of empty string?

I added a "Assignment to boolean properties" section at the bottom of my [test page](https://josepharhar.github.io/react-custom-elements/) for this.

It seems like the only difference is when we try to render the JSX attribute as `null` when the value was previously a boolean: preact will render `null` as an empty string, whereas this patch will render it as `false`.

> Seems like it's wrong either way, because it seems like we should pass null through to the property

I agree that passing `null` here makes more sense, but I'm definitely not completely aware of what use cases are at play here... would you like me to just do `null`? Or ask around why preact does an empty string and ask around which would be better?

Reply:
Is this question still relevant? The code shows up as "Outdated" so I'm not sure.

Reply:
> Is this question still relevant? The code shows up as "Outdated" so I'm not sure.

Yes - if the custom element's property value is currently `true` or `false`, and then we assign `null` into it through react, then this patch will assign `false` into it. Preact instead does an empty string, which I suppose is even more confusing, but as sebastian pointed out, we should perhaps be passing `null` through instead of `false`...

I'll try to make it just do `null` like sebastian suggested

Reply:
Ok, I pushed a commit which makes it pass `null` through and also stops using `PropertyInfo` entirely like I mentioned in my earlier comment in this thread.

I think this is resolved now, but it may be a good idea to ask for feedback on this behavior later since I haven't discussed it with anyone outside of this thread.